utils/http.go

```diff
@@ -56,14 +59,13 @@ func (root *rootHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx, w = ctxu.WithResponseWriter(ctx, w)
 	ctx = ctxu.WithLogger(ctx, log)
 	ctx = context.WithValue(ctx, notary.CtxKeyCryptoSvc, root.trust)
-
-	defer func() {
+	defer func(ctx context.Context) {
 		ctxu.GetResponseLogger(ctx).Info("response completed")
-	}()
+	}(ctx)
```

Comment:
Is this correct? Don't we want to use the possibly modified context (which may contain additional info) when the function exits?